### ISSUE TITLE
MacOS - PyTorch crashing my MAC - Pytorch compiled from Source

### ISSUE THREAD

**Issue by shankarrajus** [pytorch/pytorch#4926]
PyTorch GitHub Issues Guidelines
--------------------------------

We like to limit our issues to bug reports and feature requests. If you have a question or would like help and support, please visit our forums: https://discuss.pytorch.org/

If you are submitting a feature request, please preface the title with [feature request].

When submitting a bug report, please include the following information (where relevant):
- OS: MacOS 10.13.3
- PyTorch version: 0.4.0a0+712a6c6 
- How you installed PyTorch (conda, pip, source): source
- Python version: Python 3.6.4 |Anaconda, Inc
- CUDA/cuDNN version: 9.1 / 7
- GPU models and configuration: Nvidia GT 750M
- GCC version (if compiling from source):
_Configured with: --prefix=/Applications/Xcode8.3.3.app/Contents/Developer/usr --with-gxx-include-dir=/usr/include/c++/4.2.1
Apple LLVM version 8.1.0 (clang-802.0.42)
Target: x86_64-apple-darwin17.4.0
Thread model: posix
InstalledDir: /Applications/Xcode8.3.3.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin_

1) I am trying to run fastai lesson1.ipynb which internally uses pytorch. Since I want to make use of my GPU , I built from source torch, torchvision, text (all taken from GIT repo).
2) I tried with 10 rows and it ran fine in jupyter notebook (with following warning message)
_/Users/Shankar/GIT/fastai/courses/dl1/fastai/core.py:23: UserWarning: volatile was removed and now has no effect. Use `with torch.no_grad():` instead.
  x = Variable(T(x), volatile=volatile, requires_grad=requires_grad)_

3) Then I tried to use the entire dataset of 25K , then my mac crashed and restarted.

I did not even got error message, any idea on how to proceed further?

**Comment by apaszke** [comment]
Are you sure you're not running out of memory? `volatile` greatly decreases the usage, but was deprecated in favor of `torch.no_grad()`. Try wrapping the code that uses `x` in this context manager.

**Comment by shankarrajus** [comment]
Thanks @apaszke . I tried to run_test.sh and it failed on memory test 

_RuntimeError: unable to open shared memory object </torch_1017_822569775> in read-write mode at /Users/SSR/GIT/pytorch/aten/src/TH/THAllocator.c:290_

**Comment by apaszke** [comment]
It's likely that you see it because the system has run out of memory. How much RAM do you have on your mac?

**Comment by shankarrajus** [comment]
I have 16GB of RAM

**Comment by shankarrajus** [comment]
I ended up using ubuntu 16.04 with PyTorch 3.0 (cuda 9.0), crashing due to memory issue has stopped. But still I get "cuda run time error: out of memory issue" . I noticed, that the error happens only when my GPU has less than 1GB of free space(GPU: 2GB Nvidia GT 750M). 

So using this torch.cuda.empty_cache()  frees up few memory from GPU but not much. I had to restart my jupyter kernel(which is frustrating) to get more free memory. Is there any other way to accomplish it?
